src/BayesFilters/src/KFCorrection.cpp

```diff
@@ -99,5 +112,9 @@ void KFCorrection::correctStep(const GaussianMixture& pred_state, GaussianMixtur
         /* Evaluate the filtered covariance
            P_{k}+ = P_{k}- - K * Py * K' */
         corr_state.covariance(i).noalias() = pred_state.covariance(i) - K * Py * K.transpose();
+
+        /* Evaluate the likelihood associated to the current component.
+           This is required to implement the method getLikelihood(). */
+        likelihood_(i) = utils::multivariate_gaussian_density(innovations.col(i), VectorXd::Zero(innovations.rows()), Py).coeff(0);
```

Comment:
I would rather save the `innovations` and `Py` as class data member instead of the `likelihood`. It is quite rare that you need the likelihood from the kalman filters, but is definitely more likelihood that you need the innovation and it's likelihood. If requested by the user, we can then evaluate and return the likelihood from the latest available data.

Reply:
Agree on this :+1: